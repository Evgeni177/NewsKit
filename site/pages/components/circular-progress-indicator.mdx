import Layout from '../../components/layout';
import Playground from '../../components/playground';
import {Prop} from '../../components/prop';
import {TextBoxSwatch} from '../../components/theming-values/colors';
import {LegacyBlock} from '../../components/legacy-block';

import {CircularProgressIndicator} from 'newskit';

export default Layout;

# Circular progress indicator

## Overview

Circular progress indicators show the progression of a system operation such as downloading, uploading, processing, etc. in a visual way.

## Behaviour

Circular progress indicators display progress by animating the indicator along the length of the track. The circular indicator fill the circular track with colour, as it moves from 0 to 360 degrees over a quicker duration and repeats the animation until the action has been completed.

## Usage

Circular progress indicators display progress by animating an indicator along an invisible circular track in a clockwise direction. They can be applied to buttons, inputs or other elements.

Don't show more than one circular progress indicator as the user is likely to become overwhelmed! If more than one piece of content is processing, stick to a large for the whole screen and consider skeleton loading or linear progress indicators.

### Theming

## Props

<LegacyBlock marginTop="sizing040">
  <Prop name="$size" type="IconSizeKeys" default="iconSize020">
    Set the size of the progress indicator.
  </Prop>
  <Prop name="hideTrack" type="boolean" default={false}>
    If true the progress indicator's track is not displayed.
  </Prop>
    <Prop name="ariaLabel" type="string" default="loading">
    Used to describe the type of progress for accessibility purposes, e.g. loading.
  </Prop>
  <Prop name="$trackStylePreset" type="string" default='circleLoaderTrack010'> 
     Set the style preset of the circular track.
  </Prop>
  <Prop name="$indicatorStylePreset" type="string" default='circleLoaderIndicator010'>
     Set the style preset of the circular indicator.
  </Prop>
</LegacyBlock>

## Playground

<Playground
  componentName="CircularProgressIndicator"
  newskitPath="/circular-progress-indicator"
  component={CircularProgressIndicator}
  knobs={[
    {
      name: 'Size',
      propName: '$size',
      options: [
        {
          label: 'iconSize010',
          value: 'iconSize010',
        },
        {
          label: 'iconSize020',
          value: 'iconSize020',
          isDefault: true,
        },
        {
          label: 'iconSize030',
          value: 'iconSize030',
        },
        {
          label: 'iconSize040',
          value: 'iconSize040',
        },
        {
          label: 'iconSize050',
          value: 'iconSize050',
        },
      ],
    },
    {
      name: 'Hide Track',
      propName: 'hideTrack',
      type: 'boolean',
      value: false,
    },
    {
      name: 'trackStylePreset',
      propName: '$trackStylePreset',
      options: [
        {
          label: 'circleLoaderTrack010',
          value: 'circleLoaderTrack010',
          isDefault: true,
        },
        {
          label: 'circleLoaderTrack010Inverse',
          value: 'circleLoaderTrack010Inverse',
        },
      ],
    },
    {
      name: 'indicatorStylePreset',
      propName: '$indicatorStylePreset',
      options: [
        {
          label: 'circleLoaderIndicator010',
          value: 'circleLoaderIndicator010',
          isDefault: true,
        },
        {
          label: 'circleLoaderIndicator010Inverse',
          value: 'circleLoaderIndicator010Inverse',
        },
      ],
    },
  ]}
/>

### Accessibility

When applying a progress indicator then the aria-busy state should be applied so that users know content is being updated.
