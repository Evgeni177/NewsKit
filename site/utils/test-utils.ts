import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
