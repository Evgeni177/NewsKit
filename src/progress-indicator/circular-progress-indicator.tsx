import React from 'react';
import {CircularProgressIndicatorProps} from './types';
import {
  StyledCircularProgressIndicator,
  StyledCircularTrack,
  StyledCircularProgressIndicatorFills,
  StyledLeftHalfCircle,
  StyledLeftHalfContainer,
  StyledCircularIndicator,
  StyledRightHalfCircle,
  StyledRightHalfContainer,
} from './styled';

export const CircularProgressIndicator: React.FC<
  CircularProgressIndicatorProps
> = ({
  $size,
  hideTrack = false,
  ariaLabel = 'loading',
  $trackStylePreset,
  $indicatorStylePreset,
}) => (
  <StyledCircularProgressIndicator $size={$size} aria-label={ariaLabel}>
    {!hideTrack && (
      <StyledCircularTrack
        $size={$size}
        $trackStylePreset={$trackStylePreset}
      />
    )}

    <StyledCircularProgressIndicatorFills $size={$size}>
      <StyledLeftHalfCircle $size={$size}>
        <StyledLeftHalfContainer $size={$size}>
          <StyledCircularIndicator
            $size={$size}
            $indicatorStylePreset={$indicatorStylePreset}
          />
        </StyledLeftHalfContainer>
      </StyledLeftHalfCircle>

      <StyledRightHalfCircle $size={$size}>
        <StyledRightHalfContainer $size={$size}>
          <StyledCircularIndicator
            $size={$size}
            $indicatorStylePreset={$indicatorStylePreset}
          />
        </StyledRightHalfContainer>
      </StyledRightHalfCircle>
    </StyledCircularProgressIndicatorFills>
  </StyledCircularProgressIndicator>
);
