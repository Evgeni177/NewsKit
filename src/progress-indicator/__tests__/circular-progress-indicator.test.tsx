import {renderToFragmentWithTheme} from '../../test/test-utils';
import {CircularProgressIndicator} from '../circular-progress-indicator';

describe('Circle progress indicator', () => {
  test(`renders default circle progress indicator`, () => {
    const fragment = renderToFragmentWithTheme(CircularProgressIndicator);

    expect(fragment).toMatchSnapshot();
  });

  test(`renders circle progress indicator with size`, () => {
    const fragment = renderToFragmentWithTheme(CircularProgressIndicator, {
      $size: 'iconSize040',
    });

    expect(fragment).toMatchSnapshot();
  });

  test(`renders circle progress indicator without track`, () => {
    const fragment = renderToFragmentWithTheme(CircularProgressIndicator, {
      hideTrack: true,
    });

    expect(fragment).toMatchSnapshot();
  });

  test(`renders circle progress indicator with different style`, () => {
    const fragment = renderToFragmentWithTheme(CircularProgressIndicator, {
      $sliderTrackStylePreset: 'circularProgressIndicatorTrackInverse',
      $sliderIndicatorTrackStylePreset:
        'circularProgressIndicatorIndicatorInverse',
    });

    expect(fragment).toMatchSnapshot();
  });

  test(`renders circle progress indicator with aria label`, () => {
    const fragment = renderToFragmentWithTheme(CircularProgressIndicator, {
      ariaLabel: 'progress',
    });

    expect(fragment).toMatchSnapshot();
  });
});
