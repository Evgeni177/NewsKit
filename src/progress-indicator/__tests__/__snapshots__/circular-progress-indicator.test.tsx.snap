// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle progress indicator renders circle progress indicator with aria label 1`] = `
<DocumentFragment>
  .emotion-8 {
  height: 24px;
  width: 24px;
  position: relative;
}

.emotion-0 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #f0f0f0;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-7 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.emotion-3 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
}

.emotion-2 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
}

.emotion-1 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #0A68C1;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-6 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-5 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
  -webkit-animation: right-wobble 1.4s linear infinite;
  animation: right-wobble 1.4s linear infinite;
}

<div
    aria-label="progress"
    class="emotion-8"
  >
    <div
      class="emotion-0"
    />
    <div
      class="emotion-7"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle progress indicator renders circle progress indicator with different style 1`] = `
<DocumentFragment>
  .emotion-8 {
  height: 24px;
  width: 24px;
  position: relative;
}

.emotion-7 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.emotion-3 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
}

.emotion-2 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
}

.emotion-6 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-5 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
  -webkit-animation: right-wobble 1.4s linear infinite;
  animation: right-wobble 1.4s linear infinite;
}

.emotion-0 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: rgba(255,255,255,0.40);
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-1 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #FFFFFF;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

<div
    aria-label="loading"
    class="emotion-8"
  >
    <div
      class="emotion-0"
    />
    <div
      class="emotion-7"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle progress indicator renders circle progress indicator with size 1`] = `
<DocumentFragment>
  .emotion-7 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.emotion-8 {
  height: 48px;
  width: 48px;
  position: relative;
}

.emotion-0 {
  box-sizing: border-box;
  height: 48px;
  width: 48px;
  border-color: #f0f0f0;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-3 {
  box-sizing: border-box;
  width: calc( 48px / 2 );
  height: 48px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
}

.emotion-2 {
  width: calc( 48px / 2 );
  height: 48px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
}

.emotion-1 {
  box-sizing: border-box;
  height: 48px;
  width: 48px;
  border-color: #0A68C1;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-6 {
  box-sizing: border-box;
  width: calc( 48px / 2 );
  height: 48px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-5 {
  width: calc( 48px / 2 );
  height: 48px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
  -webkit-animation: right-wobble 1.4s linear infinite;
  animation: right-wobble 1.4s linear infinite;
}

<div
    aria-label="loading"
    class="emotion-8"
  >
    <div
      class="emotion-0"
    />
    <div
      class="emotion-7"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle progress indicator renders circle progress indicator without track 1`] = `
<DocumentFragment>
  .emotion-7 {
  height: 24px;
  width: 24px;
  position: relative;
}

.emotion-6 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.emotion-2 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
}

.emotion-1 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
}

.emotion-0 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #0A68C1;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-5 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-4 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
  -webkit-animation: right-wobble 1.4s linear infinite;
  animation: right-wobble 1.4s linear infinite;
}

<div
    aria-label="loading"
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-0"
          />
        </div>
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-0"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle progress indicator renders default circle progress indicator 1`] = `
<DocumentFragment>
  .emotion-8 {
  height: 24px;
  width: 24px;
  position: relative;
}

.emotion-0 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #f0f0f0;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-7 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  animation: fills-outer-rotate 1.4s cubic-bezier(0.25,0.78,0.48,0.89) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.emotion-3 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
}

.emotion-2 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
}

.emotion-1 {
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border-color: #0A68C1;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
}

.emotion-6 {
  box-sizing: border-box;
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  overflow: hidden;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-5 {
  width: calc( 24px / 2 );
  height: 24px;
  -webkit-transform-origin: 100% center;
  -ms-transform-origin: 100% center;
  transform-origin: 100% center;
  overflow: hidden;
  will-change: transform;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: left-wobble 1.4s linear infinite;
  animation: left-wobble 1.4s linear infinite;
  -webkit-animation: right-wobble 1.4s linear infinite;
  animation: right-wobble 1.4s linear infinite;
}

<div
    aria-label="loading"
    class="emotion-8"
  >
    <div
      class="emotion-0"
    />
    <div
      class="emotion-7"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
